code/gapfill4EssentialTasks.m

```diff
@@ -82,13 +82,19 @@
 end
 
 
-%% empty newly introduced grRules
+%% replace grRules of newly introduced reactions with "TBD"
 
 % clean the grRules associated with gap-filling reactions
 gapfilledRxns = setdiff(outModel.rxns, model.rxns);
 [~, ind] = ismember(gapfilledRxns, outModel.rxns);
 outModel.grRules(ind) = {''};
 
+% Assign "TBD" gene to the grRules which previously were non-empty in the
+% reference model. This gene name is abbreviated from "to be determined"
+[~, ind2] = ismember(gapfilledRxns, refModel.rxns);
+nonEmptyInd = getNonEmptyList(refModel.grRules(ind2));
+outModel.grRules(ind(nonEmptyInd)) = {'TBD'};
```

Comment:
I would avoid using `'TBD'` or any other unexpected (non-gene symbol or ID) value in the `grRules` field, since it could cause unforeseen problems. I would recommend just leaving them empty, and if you want to indicate that these reactions were added due to gap filling, then it can be noted in some other annotation field, such as rxnNotes or something similar.

Reply:
The GPR `TBD` (stands for `to-be-determined`) was adopted from iCEL1314 model. And the purpose is separating these gap-filled reactions from enzymatic reactions (with regular GPRs) and the other spontaneous or exchange reactions that have empty GPRs.

Reply:
Then this information should be placed in the `spontaneous` field. Even though the iCEL1314 model used the `grRules` field for this information, it is not compatible with the way that Human-GEM, RAVEN, and COBRA have defined the content of the `grRules` field.

Reply:
Why this `TBD` (or gap-filling) information should be placed in the `spontaneous` field?

Reply:
`grRules` is for gene-reaction rules, *not* for gap-filling notes and other information. A user (or function) could very easily confuse "TBD" as a gene symbol. I suggested the `spontaneous` field because you mentioned that the use of "TBD" was to differentiate these reactions from spontaneous reactions with empty GPRs, and the entire purpose of the `spontaneous` field is to indicate which reactions are enzymatic or non-enzymatic.

Another option would be to create a `rxnNotes` field or similar, where such information could be included.

Reply:
> A user (or function) could very easily confuse "TBD" as a gene symbol.

This is a valid argument, and it sounds good to me by using `rxnNotes` field to store this information.

Reply:
The inclusion of `TBD` GPR is now reverted. The gap filling information is planned to be added to `rxnNotes` field after resolving #272.